mllib/src/main/scala/org/apache/spark/mllib/feature/HashingTF.scala

```diff
@@ -96,7 +89,7 @@ class HashingTF(val numFeatures: Int) extends Serializable {
     val termFrequencies = mutable.HashMap.empty[Int, Double]
     val setTF = if (binary) (i: Int) => 1.0 else (i: Int) => termFrequencies.getOrElse(i, 0.0) + 1.0
     document.foreach { term =>
-      val i = indexOfTerm(hashAlgorithm)(term)
+      val i = indexOf(term)
```

Comment:
This is going to check the hashAlgorithm value on every call.  I'd like us to avoid that.  If needed, you could use your indexOfTerm method but keep it private.